src/components/Wallet.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import AddSigners from './AddSigners';
+import Signers from './Signers';
+import { Device } from '../types'
+import { AppState } from '../store';
+import { getWallets, selectCandidateDevicesForActiveWallet, selectActiveWallet } from '../store/wallet';
+import { Spinner } from 'reactstrap';
+import api from '../api'
+
+interface StateProps {
+  candidateDevices: Device[];
+  deviceError: Error | null;
+  activeWallet: AppState['wallet']['activeWallet'];
+}
+
+interface DispatchProps {
+  getWallets: typeof getWallets;
+}
+
+interface State {
+  deviceBeingAdded: Device | null;
+}
+
+type Props = StateProps & DispatchProps;
+
+class Wallet extends React.Component<Props, State> {
+  state: State = {
+    deviceBeingAdded: null,
+  }
+  private addSigner = async (device: Device) => {
+    if (this.props.activeWallet) {
```

Comment:
Very minor but cool typescript thing, you can do
```
if (!this.props.activeWallet) return;
```
at the beginning of a function, and from then on use `this.props.activeWallet` without worrying about the `| null` typing. And it saves you a block of indentation.